### PR TITLE
Fix mel length computation in Qwen2-Audio

### PR DESCRIPTION
**Pull request by anferico**
# What does this PR do?

1. Updates the misleading docstring in `Qwen2AudioEncoder.forward` stating that _"Qwen2Audio does not support masking of the `input_features`"_, while in fact it _does_ make use of it. This is likely a mistake made while copying the docstring over from Whisper
2. Fixes a bug that causes the last "valid" (i.e. not masked) frame of the sequence coming out of the first two convolutional layers to be masked incorrectly when `max_mel_seq_len` is an odd number

Here's a more elaborate explanation for point 2:
`max_seq_len` represents the maximum sequence length after the first 2 convolutional layers, of which only the first one compresses the input sequences (due to `stride=2`). The compression applied by the first convolutional layer can be computed using this expression (cf. [torch.nn.Conv1d](https://docs.pytorch.org/docs/stable/generated/torch.nn.Conv1d.html)): 
```
output_length = (input_length + 2*padding - dilation * (kernel_size - 1) - 1) // stride + 1
```
which, given that:
- `padding` = 1
- `dilation` = 1
- `kernel_size` = 3
- `stride` = 2

can be simplified to:

```
output_length = (input_length - 1) // 2 + 1
```

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?
@ylacombe @amyeroberts @faychu

### CODE REVIEW

**Comment by anferico** [comment]
@ylacombe @amyeroberts @faychu any news?

**Comment by Rocketknight1** [comment]
cc @eustlb @ebezzam
